### PR TITLE
buildsystem: add target debug-client

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description

We have a test setup with some boards connected to a Raspberry Pi. We want to debug remotely, not from the Pi.
The Pi can start the debug server and the host can use the newly added `debug-client` target.
Integration is only added for OpenOCD.

### Testing procedure

`RROGRAMMER=openocd BOARD=same54-xpro make debug-server`

```
### Starting GDB Server ###
Open On-Chip Debugger 0.12.0-rc3-00999-gfcb40f49b (2023-05-12-10:54)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
swd
Info : CMSIS-DAP: SWD supported
Info : CMSIS-DAP: FW Version = 03.25.01B6
Info : CMSIS-DAP: Serial# = ATML2748101800010083
Info : CMSIS-DAP: Interface Initialised (SWD)
Info : SWCLK/TCK = 1 SWDIO/TMS = 1 TDI = 1 TDO = 1 nTRST = 0 nRESET = 1
Info : CMSIS-DAP: Interface ready
Info : clock speed 2000 kHz
Info : SWD DPIDR 0x2ba01477
Info : [atsame5.cpu] Cortex-M4 r0p1 processor detected
Info : [atsame5.cpu] target has 6 breakpoints, 4 watchpoints
Info : starting gdb server for atsame5.cpu on 3333
Info : Listening on port 3333 for gdb connections
    TargetName         Type       Endian TapName            State       
--  ------------------ ---------- ------ ------------------ ------------
 0* atsame5.cpu        cortex_m   little atsame5.cpu        halted

Info : Listening on port 6333 for tcl connections
Info : Listening on port 4444 for telnet connections
Info : accepting 'gdb' connection on tcp/3333
Info : SAM MCU: SAME54P20A (1024KB Flash, 256KB RAM)
undefined debug reason 8 - target needs reset
Info : Auto-detected RTOS: RIOT
```

`BOARD=same54-xpro make debug-client`

```
### Attaching to GDB Server ###
Reading symbols from /home/fabian.huessler@ml-pa.loc/RIOT/examples/gnrc_networking/bin/same54-xpro/gnrc_networking.elf...
Remote debugging using :3333
__set_PRIMASK (priMask=1)
    at /home/fabian.huessler@ml-pa.loc/RIOT/build/pkg/cmsis/CMSIS/Core/Include/cmsis_gcc.h:1236
1236      __ASM volatile ("MSR primask, %0" : : "r" (priMask) : "memory");
(gdb)
```

### CODE REVIEW

File: dist/tools/openocd/openocd.sh [hunk 1]
```diff
@@ -50,6 +50,11 @@
 #               TELNET_PORT:    port opened for telnet connections
 #               DBG:            debugger client command, default: 'gdb -q'
 #               TUI:            if TUI!=null, the -tui option will be used
+# debug-client: debug-client <elffile>
+#               connects to a running debug-server
+#               GDB_PORT:       port opened for GDB connections
```
Comment (benpicco) [inline dist/tools/openocd/openocd.sh:55]: How can the server be specified to which this connects?

Comment (fabian18) [reply dist/tools/openocd/openocd.sh:55]: `OPENOCD_SERVER_ADDRESS` becomes part of `DBG_FLAGS` in `openocd.sh`.
```sh
: ${DBG_DEFAULT_FLAGS:=-q -ex \"tar ext ${OPENOCD_SERVER_ADDRESS}:$(( GDB_PORT + GDB_PORT_CORE_OFFSET ))\"}
: ${DBG_FLAGS:=${DBG_DEFAULT_FLAGS} ${DBG_EXTRA_FLAGS}}
```

Comment (fabian18) [reply dist/tools/openocd/openocd.sh:55]: Do you want a `GDB_ADDRESS` which is set to `OPENOCD_SERVER_ADDRESS`?

Comment (benpicco) [reply dist/tools/openocd/openocd.sh:55]: Ah no, I see that `OPENOCD_SERVER_ADDRESS` is now documented further up in this file.

**Review by benpicco — APPROVED** [review]

**Comment by benpicco** [comment]
bors merge

**Pull request merged**